packages/react/src/Stack/Stack.test.tsx

```diff
@@ -0,0 +1,254 @@
+import React from 'react'
+import {render, cleanup} from '@testing-library/react'
+import '@testing-library/jest-dom'
+import {axe, toHaveNoViolations} from 'jest-axe'
+
+import {Stack} from './Stack'
+
+expect.extend(toHaveNoViolations)
+
+describe('Stack', () => {
+  afterEach(cleanup)
+
+  const mockText = 'mock text'
+
+  it('has no a11y violations on initial render', async () => {
+    const {container} = render(
+      <Stack>
+        <></>
+      </Stack>
+    )
+    const results = await axe(container)
+
+    expect(results).toHaveNoViolations()
+  })
+
+  it('should render any type of child', () => {
+    const {getByText} = render(
+      <Stack>
+        <div>{mockText}</div>
+        <span>{1}</span>
+        <>Fragment</>
+      </Stack>
+    )
+    expect(getByText(mockText)).toBeInTheDocument()
+    expect(getByText(1)).toBeInTheDocument()
+    expect(getByText('Fragment')).toBeInTheDocument()
+  })
+
+  it('should set the correct styles for direction', () => {
+    const mockTestId = 'test'
+    const expectedClass = 'Stack--vertical'
+
+    const {getByTestId} = render(
+      <Stack direction={'vertical'} data-testid={mockTestId}>
```

Comment:
Nitpicking, sorry... can we stick to string types as there's no interpolation?

```suggestion
      <Stack direction="vertical" data-testid={mockTestId}>
```